examples/basics/emag_basics.py

```diff
@@ -3,220 +3,249 @@
+# disable all popup messages from Motor-CAD.
 #
 # Perform required imports
 # ~~~~~~~~~~~~~~~~~~~~~~~~
+# Import ``pymotorcad`` to access Motor-CAD. Import ``pyplot`` from ``matplotlib`` to display
+# graphics. Import ``os``, ``shutil``  and ``tempfile`` to open and save a temporary MOT
+# file if none is open.
 
-
+# sphinx_gallery_thumbnail_number = -2
 import os
+import shutil
+import tempfile
 
 import matplotlib.pyplot as plt
 
 import ansys.motorcad.core as pymotorcad
 
-if "QT_API" in os.environ:
-    os.environ["QT_API"] = "pyqt"
-
 # %%
 # Specify working directory
 # ~~~~~~~~~~~~~~~~~~~~~~~~~
-
-working_folder = os.getcwd()
-
-if os.path.isdir(working_folder) is False:
-    print("Working folder does not exist. Choose a folder that exists and try again.")
-    print(working_folder)
-    exit()
+# Save the file to a temporary folder
+working_folder = os.path.join(tempfile.gettempdir(), "basic_examples")
+try:
+    shutil.rmtree(working_folder)
+except:
+    pass
+os.mkdir(working_folder)
+mot_name = "EMagnetic"
 
 # %%
 # Launch Motor-CAD
 # ~~~~~~~~~~~~~~~~
-
+# Connect to Motor-CAD and open a new Motor-CAD instance. To keep a new Motor-CAD instance open
+# after executing the script, use the ``MotorCAD(keep_instance_open=True)`` option when opening the
+# new instance.
 print("Starting initialization.")
-mcad = pymotorcad.MotorCAD()
+mc = pymotorcad.MotorCAD()
 
 # %%
 # Disable popup messages
 # ~~~~~~~~~~~~~~~~~~~~~~
-
-mcad.set_variable("MessageDisplayState", 2)
-print("Initialization completed.")
-print("Running simulation.")
-
-# %%
-# Create analysis
-# ---------------
-# Creating the analysis consists of showing the magnetic context, displaying
-# the **Scripting** tab, setting the geometry and parameters, and saving
-# the file.
+# Disable all popup messages from Motor-CAD by setting the ``MessageDisplayState`` parameter to 2.
 #
-# Show the magnetic context.
-mcad.show_magnetic_context()
+# .. note::
+#    This will also disable crucial popups, for example prompts to save data or overwrite data, or
+#    dialogs used to reconcile differences in material data between the database and MOT file. In
+#    each case the default action will be taken. This setting will persist until Motor-CAD is
+#    closed.
+mc.set_variable("MessageDisplayState", 2)
 
 # %%
-# Display the **Scripting** tab.
-mcad.display_screen("Scripting")
+# Set Motor-CAD to EMag
+# ~~~~~~~~~~~~~~~~~~~~~
+# Show the E-Magnetic context in Motor-CAD and ensure that the BPM
+# Motor Type is set.
+mc.show_magnetic_context()
+mc.set_variable("Motor_Type", 0)
 
 # %%
-# Set the geometry.
-mcad.set_variable("Slot_Number", 24)
-mcad.set_variable("Tooth_Width", 6)
-mcad.set_variable("Magnet_Thickness", 4.5)
+# Set up analysis
+# ---------------
+# Setting up the analysis consists of setting the geometry, winding and material parameters, setting
+# the calculation options and operating point, and saving the file.
+#
+# Geometry setup
```

Comment:
All the section headings should be set up so that they start with a simple verb and then the action. For example, the heading here would be: Set up geometry. The heading for the next section would be: Set up winding. Please change the headings to follow this format.